04-materials/02-anatomy-of-extensions.md

```diff
@@ -77,32 +117,373 @@ graph TB
     copier copy --trust https://github.com/jupyterlab/extension-template .
     ```
 
-3. Install it in development mode
+    Be sure to select `frontend-and-server` as the "kind"!
+    Everything else can be left as default.
+
+    ![A demo of instantiating an extension project from the official template](../assets/images/init-from-template.gif)
+
+3. List out the files that were created (`ls -la` or `tree -a` are good options)
+
+4. Install the extension in development mode
 
     ```bash
     # Create and activate a virtualenv
     uv venv .venv
     source .venv/bin/activate
 
     # Install package in development mode
-    uv pip install -e ".[test,dev]"
-    # Link your development version of the extension with JupyterLab
-    # TODO: This errors unless the user has a global JupyterLab install
+    uv pip install --editable ".[dev,test]"
+
+    # Install the frontend and backend components of the extension in development mode:
     jupyter labextension develop . --overwrite
-    # Server extension must be manually installed in develop mode
     jupyter server extension enable myextension
+
     # Rebuild extension Typescript source after making changes
+    # IMPORTANT: You must do this every time you make a change!
     jlpm build
     ```
 
-4. 🧪 Test it out!
+5. 🧪 Test it out! Run this command in a **separate terminal**.
+   Remember to activate the virtual environment again with `source .venv/bin/activate`
+   any time you create a new terminal.
+   You can keep this terminal open and running JupyterLab in the background!
 
     ```bash
     jupyter lab
     ```
 
-5. Confirm the extension was loaded. Open your browser's dev console (F12 or
+6. Confirm the extension was loaded. Open your browser's dev console (F12 or
    `CTRL+SHIFT+I`) and look for log messages reading:
 
    * `JupyterLab extension myextension is activated`
    * `This is /myextension/get-example endpoint!`
+
+   **If you do not see these messages, let us know you need help!**
```

Comment:
Sidebar: maybe in the future the extension template should use Notifications API instead of console.log to be more user friendly?